libursa/benches/cks_revok.rs

```diff
@@ -0,0 +1,244 @@
+    let (credential_pub_key, credential_priv_key, credential_key_correctness_proof) = Issuer::new_credential_def(&credential_schema, &non_credential_schema, true).unwrap();
+
+    // 3. Issuer creates revocation registry
+    let (rev_key_pub, rev_key_priv, mut rev_reg, mut rev_tails_generator) =
+        Issuer::new_revocation_registry_def(&credential_pub_key, max_cred_num, issuance_by_default).unwrap();
+
+    let simple_tail_accessor = SimpleTailsAccessor::new(&mut rev_tails_generator).unwrap();
+
+    let mut _prover_data: Vec<(CredentialValues, CredentialSignature, Witness)> = vec![];
+
+    let mut rev_reg_delta: Option<RevocationRegistryDelta> = None;
+
+    for i in 0..max_cred_num {
+        let credential_values = get_credential_values(&Prover::new_master_secret().unwrap());
+
+        // 5. Issuer creates nonce used by Prover to create correctness proof for blinded secrets
+        let blinding_correctness_nonce = new_nonce().unwrap();
+
+        // 6. Prover blinds master secret
+        let (blinded_credential_secrets, credential_secrets_blinding_factors, blinded_credential_secrets_correctness_proof) =
+            Prover::blind_credential_secrets(&credential_pub_key,
+                                             &credential_key_correctness_proof,
+                                             &credential_values,
+                                             &blinding_correctness_nonce).unwrap();
+
+        // 7. Prover creates nonce used by Issuer to create correctness proof for signature
+        let signature_correctness_nonce = new_nonce().unwrap();
+
+        // 8. Issuer creates and sign credential values
+        let rev_idx = i + 1;
+        let (mut credential_signature, signature_correctness_proof, rr_delta) =
+            Issuer::sign_credential_with_revoc(&rev_idx.to_string(),
+                                               &blinded_credential_secrets,
+                                               &blinded_credential_secrets_correctness_proof,
+                                               &blinding_correctness_nonce,
+                                               &signature_correctness_nonce,
+                                               &credential_values,
+                                               &credential_pub_key,
+                                               &credential_priv_key,
+                                               rev_idx,
+                                               max_cred_num,
+                                               issuance_by_default,
+                                               &mut rev_reg,
+                                               &rev_key_priv,
+                                               &simple_tail_accessor).unwrap();
+
+
+        if i == 0 {
+            rev_reg_delta = rr_delta;
+        } else {
+            let mut _d = rev_reg_delta.unwrap();
+            _d.merge(&rr_delta.unwrap()).unwrap();
+            rev_reg_delta = Some(_d)
```

Comment:
I thought the only reason to name a variable with a leading _ was to tell the compiler that it isn't used so don't issue a warning. Here `_d` is used.

Reply:
Fixed